pkg/tsdb/v3iotsdb_integration_test.go

```diff
@@ -713,64 +732,76 @@ func TestDeleteTable(t *testing.T) {
 				t.Skip(test.ignoreReason)
 			}
 			testDeleteTSDBCase(t, v3ioConfig, "metricToDelete", utils.LabelsFromStrings("os", "linux"),
-				test.data, test.deleteFrom, test.deleteTo, test.ignoreErrors, test.expected)
+				test.data, test.deleteFrom, test.deleteTo, test.ignoreErrors, test.deleteAll, test.expected)
 		})
 	}
 }
 
 func testDeleteTSDBCase(test *testing.T, v3ioConfig *config.V3ioConfig, metricsName string, userLabels []utils.Label,
-	data []tsdbtest.DataPoint, deleteFrom int64, deleteTo int64, ignoreErrors bool,
+	data []tsdbtest.DataPoint, deleteFrom int64, deleteTo int64, ignoreErrors bool, deleteAll bool,
 	expected []tsdbtest.DataPoint) {
 
 	adapter, teardown := tsdbtest.SetUpWithData(test, v3ioConfig, metricsName, data, userLabels)
 	defer teardown()
 
 	pm, err := partmgr.NewPartitionMngr(adapter.GetSchema(), nil, v3ioConfig)
+	if err != nil {
+		test.Fatalf("Failed to create new partition manager. reason: %s", err)
+	}
 
-	initiaPartitions := pm.PartsForRange(0, time.Now().Unix()*1000, true)
+	initiaPartitions := pm.PartsForRange(0, math.MaxInt64, true)
 	initialNumberOfPartitions := len(initiaPartitions)
 
 	partitionsToDelete := pm.PartsForRange(deleteFrom, deleteTo, false)
 
-	if err := adapter.DeleteDB(false, ignoreErrors, deleteFrom, deleteTo); err != nil {
-		test.Fatalf("Failed to delete DB on teardown. reason: %s", err)
+	if err := adapter.DeleteDB(deleteAll, ignoreErrors, deleteFrom, deleteTo); err != nil {
+		test.Fatalf("Failed to delete DB. reason: %s", err)
 	}
 
-	pm1, err := partmgr.NewPartitionMngr(adapter.GetSchema(), nil, v3ioConfig)
-	remainingParts := pm1.PartsForRange(0, time.Now().Unix()*1000, false)
-	assert.Equal(test, len(remainingParts), initialNumberOfPartitions-len(partitionsToDelete))
-
-	qry, err := adapter.Querier(nil, 0, time.Now().Unix()*1000)
-	if err != nil {
-		test.Fatalf("Failed to create Querier. reason: %v", err)
-	}
+	if deleteAll {
+		_, err := NewV3ioAdapter(v3ioConfig, nil, nil)
+		if err == nil {
+			test.Fatalf("Table was deleted - failure to create adapter is expected...")
+		}
+		test.Skip("skipping tearDown since table is completely deleted")
```

Comment:
No.
You cannot skip test in the middle. Use proper Setup/TearDown implementation - override default one

Reply:
will do it tomorrow in another pr